src/protocol.rs

```diff
@@ -93,6 +93,7 @@ pub(crate) async fn read_lp(
     };
     let mut reader = reader.take(size);
     let size = usize::try_from(size).context("frame larger than usize")?;
+    buffer.reserve(size);
```

Comment:
This should be limited, I can now easily attack your machine by putting a crazy value here, and causing an OOM

Reply:
My thinking was that I wasn't at all changing how much is being allocated, this attack already existed.  To be fair I also faultily assumed MAX_MESSAGE_SIZE was already being respected while it obviously wasn't.

So fair shout, MAX_MESSAGE_SIZE wasn't being checked yet so may as well improve this.